csm/common/timeseries.py

```diff
@@ -270,15 +270,20 @@ async def _parse_interval(self, from_t, duration_t, interval, total_sample):
         duration_t = int(duration_t) - int(self._offset_interval)
         if total_sample == "" and interval == "":
             interval = str(self._storage_interval) + 's'
-        elif total_sample != "":
+        elif total_sample != "" and interval == "":
```

Comment:
This check is not necessary here -- > interval == "" . If total_sample is given then `interval` query parameter will not come into the picture, so any validation should be skipped .

Reply:
Addressed